code/modules/shuttle/arrivals.dm

```diff
@@ -0,0 +1,145 @@
+/obj/docking_port/mobile/arrivals
+	name = "arrivals shuttle"
+	id = "arrivals"
+
+	dwidth = 3
+	width = 7
+	height = 15
+	dir = WEST
+	preferred_direction = WEST
+	port_angle = 180
+	ignitionTime = 50
+
+	roundstart_move = TRUE	//force a call to dockRoundstart
+
+	var/docked	//docked at arrivals?
+	var/damaged	//too damaged to undock?
+	var/list/areas	//areas in our shuttle
+	var/list/queued_announces	//people coming in that we have to announce
+	var/roundstart_docked
+
+/obj/docking_port/mobile/arrivals/Initialize(mapload)
+	if(mapload)
+		return TRUE	//late initialize to make sure the latejoin list is populated
+
+	..()
+
+	if(SSshuttle.arrivals)
+		WARNING("More than one arrivals docking_port placed on map!")
+		qdel(src)
+		return
+
+	SSshuttle.arrivals = src
+
+	areas = list()
+
+	if(latejoin.len)
+		WARNING("Map contains predefined latejoin spawn points and an arrivals shuttle. Using the arrivals shuttle.")
+
+	latejoin = list()
+	for(var/area/shuttle/arrival/A in world)
+		for(var/obj/structure/chair/C in A)
+			latejoin += C
+		areas += A
+
+/obj/docking_port/mobile/arrivals/dockRoundstart()
+	SSshuttle.generate_transit_dock(src)
+	. = dock(assigned_transit)
+	roundstart_docked = TRUE
+
+/obj/docking_port/mobile/arrivals/check()
+	. = ..()
+
+	var/docked = src.docked
+
+	if(damaged)
+		//TODO: repair checks
+		return
+	//TODO: check for damage

```

Comment:
instead of marooning the shuttle if it's broken, just have it get repaired (a la, load from template) when it has a chance